pkg/snowflake/builder.go

```diff
@@ -0,0 +1,419 @@
+package snowflake
+
+import (
+	"database/sql"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+type ParamType string
+
+var (
+	Integer    ParamType = "int"
+	String     ParamType = "string"
+	StringList ParamType = "stringlist"
+)
+
+type SQLParameter struct {
+	name      string
+	paramType ParamType
+}
+
+type BuilderConfig struct {
+	beforeObjectType map[string]string
+	afterObjectType  map[string]string
+	parameters       map[string]*SQLParameter
+}
+
+type NewBuilder struct {
+	objectType       string
+	objectTypePlural string
+	createConfig     *BuilderConfig
```

Comment:
maybe make a nested struct for configs. since you have so many of them. also why even does the config need to be saved? couldn't it be purely functional?